FreeRTOS/FreeRTOS-Kernel
# Add configUSE_MINI_LIST_ITEM configuration option to enable the MiniListItem_t type.

Description
-----------
Add configUSE_MINI_LIST_ITEM configuration option to enable the MiniListItem_t type.
    
When configUSE_MINI_LIST_ITEM == 0:
        MiniListItem_t and ListItem_t are both typedefs of struct xLIST_ITEM.

When configUSE_MINI_LIST_ITEM == 1 (the default in projdefs.h):
        MiniListItem_t is a typedef of struct xMINI_LIST_ITEM, which contains 3 fewer fields than a struct xLIST_ITEM.
        This configuration saves approximately sizeof(TickType_t) + 2 * sizeof( void * ) bytes of ram, however it also violates strict aliasing rules which some compilers depend on for optimization.

configUSE_MINI_LIST_ITEM defaults to 1 when not defined.

Test Steps
-----------
Run unit tests.
Compile with both settings for configUSE_MINI_LIST_ITEM.

Related Issue
-----------
https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/53

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
